js/site.js

```diff
@@ -20,12 +40,19 @@ class Site{
     getScore(){
         this.score = this.scoreFunction();
         return this.score;
-    }
+    };
 
-    getTrackers(){ return this.trackers };
+    setWhitelistStatusFromGlobal(domain){
+        let globalWhitelist = settings.getSetting('whitelist');
+        if(globalWhitelist && globalWhitelist[this.domain]){
```

Comment:
I think this expression is not right.

if `settings.getSetting('whitelist');` is null you want to create it, right? Seems odd that you should make the site object responsible for creating the global whitelist.

isn't it possible that this expression evaluates to false when `globalWhitelist[this.domain]` is true, in which case you want to create and set?